client/my-sites/domains/domain-management/edit/domain-types/mapped-domain-type.jsx

```diff
@@ -0,0 +1,360 @@
+							},
+						}
+					) }
+				</p>
+				<RenewButton
+					primary={ true }
+					selectedSite={ this.props.selectedSite }
+					subscriptionId={ parseInt( domain.subscriptionId, 10 ) }
+					tracksProps={ { source: 'mapped-domain-status', mapping_status: 'expiring-soon' } }
+				/>
+			</div>
+		);
+	}
+
+	renderSettingUpNameservers() {
+		const { domain, selectedSite, translate } = this.props;
+		if (
+			selectedSite.jetpack ||
+			( selectedSite.options && selectedSite.options.is_automated_transfer )
+		) {
+			return null;
+		}
+
+		if ( domain.pointsToWpcom ) {
+			return null;
+		}
+
+		const learnMoreLink = linksTo => (
+			<a href={ linksTo } target="_blank" rel="noopener noreferrer" />
+		);
+		let primaryMessage;
+		let secondaryMessage;
+
+		if ( isSubdomain( domain.name ) ) {
+			primaryMessage = translate(
+				'Your subdomain mapping has not been setup. You need to create the correct CNAME or NS records at your current DNS provider. {{learnMoreLink}}Learn how to do that in our support guide for mapping subdomains{{/learnMoreLink}}.',
+				{
+					components: {
+						strong: <strong />,
+						learnMoreLink: learnMoreLink( MAP_SUBDOMAIN ),
+					},
+					args: { domainName: domain.name },
+					context: 'Notice for mapped subdomain that has DNS records need to set up',
+				}
+			);
+			secondaryMessage = translate(
+				"Please note that it can take up to 72 hours for your changes to become available. If you're still not seeing your site loading at %(domainName)s, please wait a few more hours, clear your browser cache, and try again.",
+				{
+					args: { domainName: domain.name },
+				}
+			);
+		} else {
+			primaryMessage = translate(
+				'Your domain mapping has not been setup. You need to update your nameservers, at the company you purchased your domain, to:',
+				{
+					context: 'Notice for mapped domain notice with NS records pointing to somewhere else',
+				}
+			);
+			secondaryMessage = translate(
+				"Please note that it can take up to 72 hours for your changes to become available. If you're still not seeing your site loading at %(domainName)s, please wait a few more hours, clear your browser cache, and try again. {{learnMoreLink}}Learn all about mapping existing domain in our support docs{{/learnMoreLink}}.",
+				{
+					components: { learnMoreLink: learnMoreLink( MAP_EXISTING_DOMAIN ) },
+					args: { domainName: domain.name },
+				}
+			);
+		}
+
+		return (
+			<React.Fragment>
+				<div>
+					<p>{ primaryMessage }</p>
+					{ ! isSubdomain( domain.name ) && (
+						<ul>
+							{ WPCOM_DEFAULTS.map( nameServer => {
+								return <li key={ nameServer }>{ nameServer }</li>;
+							} ) }
+						</ul>
+					) }
+				</div>
+				<div>{ secondaryMessage }</div>
+			</React.Fragment>
+		);
+	}
+
+	renderDefaultRenewButton() {
+		const { domain, translate } = this.props;
+
+		if ( domain.expired || isExpiringSoon( domain, 30 ) ) {
+			return null;
+		}
+
+		if ( domain.bundledPlanSubscriptionId ) {
```

Comment:
Same point as above about avoid duplicating the render definitions of components.